servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/DefaultGrpcClientCallFactory.java

```diff
@@ -154,13 +154,14 @@
         final BlockingStreamingClientCall<Req, Resp> streamingClientCall =
                 newBlockingStreamingCall(serializationProvider, requestClass, responseClass);
         return (metadata, request) -> {
-            final BlockingIterator<Resp> iterator = streamingClientCall.request(metadata, request).iterator();
-            final Resp firstItem = iterator.next();
-            if (iterator.hasNext()) {
-                throw new IllegalArgumentException("Only a single item expected, but saw the second value: " +
-                        iterator.next());
+            try (BlockingIterator<Resp> iterator = streamingClientCall.request(metadata, request).iterator()) {
+                final Resp firstItem = requireNonNull(iterator.next(), "Response item is null");
+                if (iterator.hasNext()) {
+                    iterator.next(); // Consume the next item to make sure it's not a TerminalNotification with an error
+                    throw new IllegalStateException("Only a single response item is expected, but saw the second one");
```

Comment:
Not changed in this PR but as you are changing code; this would be a better message.

```suggestion
                    throw new IllegalStateException("More than one response message received.");
```